src/Native/Runtime/thread.cpp

```diff
@@ -1149,6 +1164,22 @@ void Thread::ReversePInvokeAttachOrTrapThread(ReversePInvokeFrame * pFrame)
     }
 }
 
+void Thread::EnsureRuntimeInitialized()
+{
+    while (PalInterlockedCompareExchangePointer(&g_RuntimeInitializingThread, this, NULL) != NULL)
+    {
+        PalSleep(1);
+    }
+
+    if (g_RuntimeInitializationCallback != NULL)
+    {
+        g_RuntimeInitializationCallback();
```

Comment:
The callback returns `int`, but the return value is not checked. Change the return value of the callback to `void`?

Reply:
Or keep it as `int`, check the return and fail fast it is non-zero.

Reply:
Going with the second option, makes sense to act when runtime initialization fails